node-gui/backend/src/messages.rs

```diff
@@ -149,11 +150,35 @@ pub struct SendDelegateToAddressRequest {
     pub delegation_id: String,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Transaction {
+    #[serde(with = "hex_encoded_serialization")]
+    tx: SignedTransaction,
+}
+
+impl Transaction {
+    pub fn new(tx: SignedTransaction) -> Self {
+        Self { tx }
+    }
+
+    pub fn take_tx(self) -> SignedTransaction {
+        self.tx
+    }
+
+    pub fn text_summary(&self, config: &ChainConfig) -> String {
+        self.tx.transaction().text_summary(config)
+    }
+
+    pub fn to_json(&self, config: &ChainConfig) -> Result<serde_json::Value, BackendError> {
+        common::address::dehexify::to_dehexified_json(config, self.tx.transaction())
+            .map_err(|e| BackendError::JsonError(e.to_string()))
+    }
```

Comment:
I kind of missed the fact that `Transaction` is serializable. So now I'm wondering why `SignedTransaction` shouldn't be

Reply:
I don't know, my guess would be that we don't have a use case for serializing witnesses